server/app/survey.go

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package app
+
+import (
+	"fmt"
+
+	"github.com/pkg/errors"
+
+	"github.com/mattermost/mattermost-plugin-user-survey/server/model"
+)
+
+func (a *UserSurveyApp) SaveSurvey(survey *model.Survey) error {
+	survey.SetDefaults()
+	if err := survey.IsValid(); err != nil {
+		return errors.Wrap(err, "SaveSurvey: survey is not valid")
+	}
+
+	return a.store.SaveSurvey(survey)
+}
+
+func (a *UserSurveyApp) GetInProgressSurvey() (*model.Survey, error) {
+	surveys, err := a.store.GetSurveysByStatus(model.SurveyStatusInProgress)
+	if err != nil {
+		return nil, errors.Wrap(err, "GetInProgressSurvey: failed to get in progress surveys from database")
+	}
+
+	if len(surveys) > 1 {
+		return nil, fmt.Errorf("more than one in-progress survey found in the database. There should only be one in-progress survey, in_progress_survey_count: %d", len(surveys))
+	}
```

Comment:
Don't we want to give more information here? Maybe the surveys ids so a survey can be stopped? 0/5

Reply:
Done!